src/test/java/ch/wisv/areafiftylan/UserRestIntegrationTest.java

```diff
@@ -608,24 +607,46 @@ public void createProfileEmptyDisplayName() {
 
         //@formatter:off
         given().
+                filter(sessionFilter).
+                header(login.getCsrfHeader()).
+                when().
+                content(profileDTO).
+                contentType(ContentType.JSON).
+                post("/users/current/profile").
+                then().
+                statusCode(HttpStatus.SC_OK).
+                body("object.birthday", equalTo("2000-01-01")).
+                body("object.gender", is("MALE")).
+                body("object.address", equalTo("Testaddress")).
+                body("object.zipcode", equalTo("Testzipcode")).
+                body("object.city", equalTo("Testcity")).
+                body("object.phoneNumber", equalTo("TestphoneNumber")).
+                body("object.notes", equalTo("Testnotes")).
+                body("object.firstName", equalTo("TestfirstName")).
+                body("object.lastName", equalTo("TestlastName")).
+                body("object.displayName", equalTo(""));
+        //@formatter:on
+    }
+
+    @Test
+    public void createProfileDuplicateDisplayName() {
+        createEnabledTestUser();
+
+        Map<String, String> profileDTO = getProfileDTO();
+        profileDTO.put("displayName", "MONSTERKILLER9001");
+
+        SessionData login = login(testUser.getUsername(), testUserCleartextPassword);
+
+        //@formatter:off
+        given().
             filter(sessionFilter).
             header(login.getCsrfHeader()).
         when().
             content(profileDTO).
             contentType(ContentType.JSON).
             post("/users/current/profile").
         then().
-            statusCode(HttpStatus.SC_OK).
-            body("object.birthday", equalTo("2000-01-01")).
-            body("object.gender", is("MALE")).
-            body("object.address", equalTo("Testaddress")).
-            body("object.zipcode", equalTo("Testzipcode")).
-            body("object.city", equalTo("Testcity")).
-            body("object.phoneNumber", equalTo("TestphoneNumber")).
-            body("object.notes", equalTo("Testnotes")).
-            body("object.firstName", equalTo("TestfirstName")).
-            body("object.lastName", equalTo("TestlastName")).
-            body("object.displayName", equalTo(""));
+            statusCode(HttpStatus.SC_CONFLICT);
```

Comment:
Also check for the message here, as this will most likely be used in the front-end